types/react-native/index.d.ts

```diff
@@ -8976,10 +8986,10 @@ export const PixelRatio: PixelRatioStatic;
  *   const View = requireNativeComponent('RCTView');
  *
  * The concrete return type of `requireNativeComponent` is a string, but the declared type is
- * `any` because TypeScript assumes anonymous JSX intrinsics (`string` instead of `"div", for
- * example) not to have any props.
+ * `HostComponent` because TypeScript assumes anonymous JSX intrinsics (e.g. a `string`) not
+ * to have any props.
  */
-export function requireNativeComponent(viewName: string): any;
+export function requireNativeComponent<T>(viewName: string): HostComponent<T>;
```

Comment:
There is also a new codegenNativeComponent function that has the same type. It’s currently only used in core, but the method exists in 0.62 so it is reasonable to include here.

There is also a codegenNativeCommands function in core.

Actually, I’m not sure if we export either of these from the public API yet. Can you check? If we don’t export them then we shouldn’t include them here yet.

Reply:
`requireNativeComponent` is exported from the root `index.js` module, `codegenNativeCommands` is not.